platform/Stream.cpp

```diff
@@ -112,14 +103,14 @@ void Stream::rewind() {
 }
 
 int Stream::isatty() {
-    return 0;
+    return 1;
```

Comment:
A stream is attached to a device. If set to 0, stream is not treated as a device that causes troubles for people using Serial(). There would hardly be any users of this API. So I don't think it will cause any breakage. Previously isatty() was not doing anything (in mbed_retarget.cpp). We started using it to decide upon when to use unbuffered stream and so forth.